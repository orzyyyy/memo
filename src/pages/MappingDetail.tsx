import React from 'react';
import './css/mapping.css';
import { Canvas, Toolbar } from 'mini-xmind';
import { DataSource } from 'mini-xmind/lib/canvas/core';

export interface MappingDetailProps {
  dataSource: DataSource;
  onChange: (data: DataSource) => void;
}

const MappingDetail = (props: MappingDetailProps) => (
  <div className="mapping">
    <Toolbar />
    <Canvas
      data={props.dataSource}
      className="canvas-wrapper"
      onChange={props.onChange}
    />
  </div>
);

export default MappingDetail;
