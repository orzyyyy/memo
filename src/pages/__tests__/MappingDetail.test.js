import React from 'react';
import { mount } from 'enzyme';
import MappingDetail from '../MappingDetail';
import 'nino-cli/scripts/setup';

const dataSource = {
  CanvasPosition: { x: -67, y: 230 },
  BlockGroup: {
    'block-623187': {
      x: 158,
      y: 256,
    },
    'block-624018': {
      x: 367,
      y: 368,
    },
    'block-73377': {
      x: 253,
      y: 525,
    },
  },
  TagGroup: {
    'tag-626505': {
      x: 167,
      y: 284,
      style: { width: 100, height: 32 },
      editable: false,
      input: 'test',
    },
    'tag-629962': {
      x: 405,
      y: 398,
      style: { width: 100, height: 32 },
      editable: false,
      input: 'test2',
    },
    'tag-80986': {
      x: 286,
      y: 555,
      style: { width: 100, height: 32 },
      editable: false,
      input: 'test3',
    },
  },
  LineGroup: {
    'line-77619': { fromKey: 'block-73377', toKey: 'block-623187' },
    'line-592694': { fromKey: 'block-623187', toKey: 'block-624018' },
  },
};

describe('MappingDetail', () => {
  it('render correctly', () => {
    const wrapper = mount(
      <MappingDetail dataSource={dataSource} onSave={null} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
