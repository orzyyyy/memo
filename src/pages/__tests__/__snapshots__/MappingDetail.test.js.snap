// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MappingDetail render correctly 1`] = `
<MappingDetail
  dataSource={
    Object {
      "BlockGroup": Object {
        "block-623187": Object {
          "x": 158,
          "y": 256,
        },
        "block-624018": Object {
          "x": 367,
          "y": 368,
        },
        "block-73377": Object {
          "x": 253,
          "y": 525,
        },
      },
      "CanvasPosition": Object {
        "x": -67,
        "y": 230,
      },
      "LineGroup": Object {
        "line-592694": Object {
          "fromKey": "block-623187",
          "toKey": "block-624018",
        },
        "line-77619": Object {
          "fromKey": "block-73377",
          "toKey": "block-623187",
        },
      },
      "TagGroup": Object {
        "tag-626505": Object {
          "editable": false,
          "input": "test",
          "style": Object {
            "height": 32,
            "width": 100,
          },
          "x": 167,
          "y": 284,
        },
        "tag-629962": Object {
          "editable": false,
          "input": "test2",
          "style": Object {
            "height": 32,
            "width": 100,
          },
          "x": 405,
          "y": 398,
        },
        "tag-80986": Object {
          "editable": false,
          "input": "test3",
          "style": Object {
            "height": 32,
            "width": 100,
          },
          "x": 286,
          "y": 555,
        },
      },
    }
  }
  onSave={null}
>
  <div
    className="mapping"
  >
    <Toolbar>
      <div
        className="Toolbar"
      >
        <ul>
          <li
            draggable={true}
            onDragStart={[Function]}
          >
            <Minus>
              <AntdIcon
                icon={
                  Object {
                    "icon": Object {
                      "attrs": Object {
                        "focusable": "false",
                        "viewBox": "64 64 896 896",
                      },
                      "children": Array [
                        Object {
                          "attrs": Object {
                            "d": "M872 474H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h720c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z",
                          },
                          "tag": "path",
                        },
                      ],
                      "tag": "svg",
                    },
                    "name": "minus",
                    "theme": "outline",
                  }
                }
              >
                <span
                  aria-label="minus"
                  className="anticon anticon-minus"
                  role="img"
                >
                  <IconReact
                    className=""
                    icon={
                      Object {
                        "icon": Object {
                          "attrs": Object {
                            "focusable": "false",
                            "viewBox": "64 64 896 896",
                          },
                          "children": Array [
                            Object {
                              "attrs": Object {
                                "d": "M872 474H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h720c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z",
                              },
                              "tag": "path",
                            },
                          ],
                          "tag": "svg",
                        },
                        "name": "minus",
                        "theme": "outline",
                      }
                    }
                  >
                    <svg
                      aria-hidden="true"
                      className=""
                      data-icon="minus"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      key="svg-minus"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M872 474H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h720c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
                        key="svg-minus-svg-0"
                      />
                    </svg>
                  </IconReact>
                </span>
              </AntdIcon>
            </Minus>
          </li>
          <li
            draggable={true}
            onDragStart={[Function]}
          >
            <Border>
              <AntdIcon
                icon={
                  Object {
                    "icon": Object {
                      "attrs": Object {
                        "focusable": "false",
                        "viewBox": "64 64 896 896",
                      },
                      "children": Array [
                        Object {
                          "attrs": Object {
                            "d": "M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z",
                          },
                          "tag": "path",
                        },
                      ],
                      "tag": "svg",
                    },
                    "name": "border",
                    "theme": "outline",
                  }
                }
              >
                <span
                  aria-label="border"
                  className="anticon anticon-border"
                  role="img"
                >
                  <IconReact
                    className=""
                    icon={
                      Object {
                        "icon": Object {
                          "attrs": Object {
                            "focusable": "false",
                            "viewBox": "64 64 896 896",
                          },
                          "children": Array [
                            Object {
                              "attrs": Object {
                                "d": "M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z",
                              },
                              "tag": "path",
                            },
                          ],
                          "tag": "svg",
                        },
                        "name": "border",
                        "theme": "outline",
                      }
                    }
                  >
                    <svg
                      aria-hidden="true"
                      className=""
                      data-icon="border"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      key="svg-border"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
                        key="svg-border-svg-0"
                      />
                    </svg>
                  </IconReact>
                </span>
              </AntdIcon>
            </Border>
          </li>
          <li
            draggable={true}
            onDragStart={[Function]}
          >
            <Tag>
              <AntdIcon
                icon={
                  Object {
                    "icon": Object {
                      "attrs": Object {
                        "focusable": "false",
                        "viewBox": "64 64 896 896",
                      },
                      "children": Array [
                        Object {
                          "attrs": Object {
                            "d": "M938 458.8l-29.6-312.6c-1.5-16.2-14.4-29-30.6-30.6L565.2 86h-.4c-3.2 0-5.7 1-7.6 2.9L88.9 557.2a9.96 9.96 0 000 14.1l363.8 363.8c1.9 1.9 4.4 2.9 7.1 2.9s5.2-1 7.1-2.9l468.3-468.3c2-2.1 3-5 2.8-8zM459.7 834.7L189.3 564.3 589 164.6 836 188l23.4 247-399.7 399.7zM680 256c-48.5 0-88 39.5-88 88s39.5 88 88 88 88-39.5 88-88-39.5-88-88-88zm0 120c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32z",
                          },
                          "tag": "path",
                        },
                      ],
                      "tag": "svg",
                    },
                    "name": "tag",
                    "theme": "outline",
                  }
                }
              >
                <span
                  aria-label="tag"
                  className="anticon anticon-tag"
                  role="img"
                >
                  <IconReact
                    className=""
                    icon={
                      Object {
                        "icon": Object {
                          "attrs": Object {
                            "focusable": "false",
                            "viewBox": "64 64 896 896",
                          },
                          "children": Array [
                            Object {
                              "attrs": Object {
                                "d": "M938 458.8l-29.6-312.6c-1.5-16.2-14.4-29-30.6-30.6L565.2 86h-.4c-3.2 0-5.7 1-7.6 2.9L88.9 557.2a9.96 9.96 0 000 14.1l363.8 363.8c1.9 1.9 4.4 2.9 7.1 2.9s5.2-1 7.1-2.9l468.3-468.3c2-2.1 3-5 2.8-8zM459.7 834.7L189.3 564.3 589 164.6 836 188l23.4 247-399.7 399.7zM680 256c-48.5 0-88 39.5-88 88s39.5 88 88 88 88-39.5 88-88-39.5-88-88-88zm0 120c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32z",
                              },
                              "tag": "path",
                            },
                          ],
                          "tag": "svg",
                        },
                        "name": "tag",
                        "theme": "outline",
                      }
                    }
                  >
                    <svg
                      aria-hidden="true"
                      className=""
                      data-icon="tag"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      key="svg-tag"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M938 458.8l-29.6-312.6c-1.5-16.2-14.4-29-30.6-30.6L565.2 86h-.4c-3.2 0-5.7 1-7.6 2.9L88.9 557.2a9.96 9.96 0 000 14.1l363.8 363.8c1.9 1.9 4.4 2.9 7.1 2.9s5.2-1 7.1-2.9l468.3-468.3c2-2.1 3-5 2.8-8zM459.7 834.7L189.3 564.3 589 164.6 836 188l23.4 247-399.7 399.7zM680 256c-48.5 0-88 39.5-88 88s39.5 88 88 88 88-39.5 88-88-39.5-88-88-88zm0 120c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32z"
                        key="svg-tag-svg-0"
                      />
                    </svg>
                  </IconReact>
                </span>
              </AntdIcon>
            </Tag>
          </li>
        </ul>
      </div>
    </Toolbar>
    <Canvas
      className="canvas-wrapper"
      data={
        Object {
          "BlockGroup": Object {},
          "CanvasPosition": Object {
            "x": -67,
            "y": 230,
          },
          "LineGroup": Object {},
          "TagGroup": Object {
            "tag-626505": Object {
              "editable": false,
              "input": "test",
              "style": Object {
                "height": 32,
                "width": 100,
              },
              "x": 167,
              "y": 284,
            },
            "tag-629962": Object {
              "editable": false,
              "input": "test2",
              "style": Object {
                "height": 32,
                "width": 100,
              },
              "x": 405,
              "y": 398,
            },
            "tag-80986": Object {
              "editable": false,
              "input": "test3",
              "style": Object {
                "height": 32,
                "width": 100,
              },
              "x": 286,
              "y": 555,
            },
          },
        }
      }
      onChange={[Function]}
      orientation="v"
      style={Object {}}
    >
      <Draggable
        allowAnyClick={false}
        axis="both"
        bounds={false}
        cancel={null}
        defaultClassName="react-draggable"
        defaultClassNameDragged="react-draggable-dragged"
        defaultClassNameDragging="react-draggable-dragging"
        defaultPosition={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        disabled={false}
        enableUserSelectHack={true}
        grid={null}
        handle={null}
        offsetParent={null}
        onDrag={[Function]}
        onMouseDown={[Function]}
        onStart={[Function]}
        onStop={[Function]}
        position={
          Object {
            "x": -67,
            "y": 230,
          }
        }
        scale={1}
        transform={null}
      >
        <DraggableCore
          allowAnyClick={false}
          axis="both"
          bounds={false}
          cancel={null}
          defaultClassName="react-draggable"
          defaultClassNameDragged="react-draggable-dragged"
          defaultClassNameDragging="react-draggable-dragging"
          defaultPosition={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          disabled={false}
          enableUserSelectHack={true}
          grid={null}
          handle={null}
          offsetParent={null}
          onDrag={[Function]}
          onMouseDown={[Function]}
          onStart={[Function]}
          onStop={[Function]}
          position={
            Object {
              "x": -67,
              "y": 230,
            }
          }
          scale={1}
          transform={null}
        >
          <div
            className="Canvas canvas-wrapper react-draggable"
            data={
              Object {
                "BlockGroup": Object {},
                "CanvasPosition": Object {
                  "x": -67,
                  "y": 230,
                },
                "LineGroup": Object {},
                "TagGroup": Object {
                  "tag-626505": Object {
                    "editable": false,
                    "input": "test",
                    "style": Object {
                      "height": 32,
                      "width": 100,
                    },
                    "x": 167,
                    "y": 284,
                  },
                  "tag-629962": Object {
                    "editable": false,
                    "input": "test2",
                    "style": Object {
                      "height": 32,
                      "width": 100,
                    },
                    "x": 405,
                    "y": 398,
                  },
                  "tag-80986": Object {
                    "editable": false,
                    "input": "test3",
                    "style": Object {
                      "height": 32,
                      "width": 100,
                    },
                    "x": 286,
                    "y": 555,
                  },
                },
              }
            }
            onDragOver={[Function]}
            onDrop={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            onWheel={[Function]}
            style={
              Object {
                "touchAction": "none",
                "transform": "translate(-67px,230px)",
              }
            }
            transform={null}
          >
            <BlockGroup
              data={Object {}}
              lineData={Object {}}
              offset={
                Object {
                  "x": -67,
                  "y": 230,
                }
              }
              onChange={[Function]}
              onContextMenu={[Function]}
              renderLine={[Function]}
            >
              <LineGroup
                data={Object {}}
                offset={
                  Object {
                    "x": -67,
                    "y": 230,
                  }
                }
                orientation="v"
              />
            </BlockGroup>
            <TagGroup
              data={
                Object {
                  "tag-626505": Object {
                    "editable": false,
                    "input": "test",
                    "style": Object {
                      "height": 32,
                      "width": 100,
                    },
                    "x": 167,
                    "y": 284,
                  },
                  "tag-629962": Object {
                    "editable": false,
                    "input": "test2",
                    "style": Object {
                      "height": 32,
                      "width": 100,
                    },
                    "x": 405,
                    "y": 398,
                  },
                  "tag-80986": Object {
                    "editable": false,
                    "input": "test3",
                    "style": Object {
                      "height": 32,
                      "width": 100,
                    },
                    "x": 286,
                    "y": 555,
                  },
                }
              }
              onChange={[Function]}
              onContextMenu={[Function]}
            >
              <Draggable
                allowAnyClick={false}
                axis="both"
                bounds={false}
                cancel={null}
                defaultClassName="react-draggable"
                defaultClassNameDragged="react-draggable-dragged"
                defaultClassNameDragging="react-draggable-dragging"
                defaultPosition={
                  Object {
                    "x": 0,
                    "y": 0,
                  }
                }
                disabled={false}
                enableUserSelectHack={true}
                grid={null}
                handle={null}
                key="tag-626505"
                offsetParent={null}
                onDrag={[Function]}
                onMouseDown={[Function]}
                onStart={[Function]}
                onStop={[Function]}
                position={
                  Object {
                    "x": 167,
                    "y": 284,
                  }
                }
                scale={1}
                transform={null}
              >
                <DraggableCore
                  allowAnyClick={false}
                  axis="both"
                  bounds={false}
                  cancel={null}
                  defaultClassName="react-draggable"
                  defaultClassNameDragged="react-draggable-dragged"
                  defaultClassNameDragging="react-draggable-dragging"
                  defaultPosition={
                    Object {
                      "x": 0,
                      "y": 0,
                    }
                  }
                  disabled={false}
                  enableUserSelectHack={true}
                  grid={null}
                  handle={null}
                  offsetParent={null}
                  onDrag={[Function]}
                  onMouseDown={[Function]}
                  onStart={[Function]}
                  onStop={[Function]}
                  position={
                    Object {
                      "x": 167,
                      "y": 284,
                    }
                  }
                  scale={1}
                  transform={null}
                >
                  <div
                    className="tag-group animate-appear react-draggable"
                    onContextMenu={[Function]}
                    onDoubleClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    style={
                      Object {
                        "touchAction": "none",
                        "transform": "translate(167px,284px)",
                      }
                    }
                    transform={null}
                  >
                    test
                  </div>
                </DraggableCore>
              </Draggable>
              <Draggable
                allowAnyClick={false}
                axis="both"
                bounds={false}
                cancel={null}
                defaultClassName="react-draggable"
                defaultClassNameDragged="react-draggable-dragged"
                defaultClassNameDragging="react-draggable-dragging"
                defaultPosition={
                  Object {
                    "x": 0,
                    "y": 0,
                  }
                }
                disabled={false}
                enableUserSelectHack={true}
                grid={null}
                handle={null}
                key="tag-629962"
                offsetParent={null}
                onDrag={[Function]}
                onMouseDown={[Function]}
                onStart={[Function]}
                onStop={[Function]}
                position={
                  Object {
                    "x": 405,
                    "y": 398,
                  }
                }
                scale={1}
                transform={null}
              >
                <DraggableCore
                  allowAnyClick={false}
                  axis="both"
                  bounds={false}
                  cancel={null}
                  defaultClassName="react-draggable"
                  defaultClassNameDragged="react-draggable-dragged"
                  defaultClassNameDragging="react-draggable-dragging"
                  defaultPosition={
                    Object {
                      "x": 0,
                      "y": 0,
                    }
                  }
                  disabled={false}
                  enableUserSelectHack={true}
                  grid={null}
                  handle={null}
                  offsetParent={null}
                  onDrag={[Function]}
                  onMouseDown={[Function]}
                  onStart={[Function]}
                  onStop={[Function]}
                  position={
                    Object {
                      "x": 405,
                      "y": 398,
                    }
                  }
                  scale={1}
                  transform={null}
                >
                  <div
                    className="tag-group animate-appear react-draggable"
                    onContextMenu={[Function]}
                    onDoubleClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    style={
                      Object {
                        "touchAction": "none",
                        "transform": "translate(405px,398px)",
                      }
                    }
                    transform={null}
                  >
                    test2
                  </div>
                </DraggableCore>
              </Draggable>
              <Draggable
                allowAnyClick={false}
                axis="both"
                bounds={false}
                cancel={null}
                defaultClassName="react-draggable"
                defaultClassNameDragged="react-draggable-dragged"
                defaultClassNameDragging="react-draggable-dragging"
                defaultPosition={
                  Object {
                    "x": 0,
                    "y": 0,
                  }
                }
                disabled={false}
                enableUserSelectHack={true}
                grid={null}
                handle={null}
                key="tag-80986"
                offsetParent={null}
                onDrag={[Function]}
                onMouseDown={[Function]}
                onStart={[Function]}
                onStop={[Function]}
                position={
                  Object {
                    "x": 286,
                    "y": 555,
                  }
                }
                scale={1}
                transform={null}
              >
                <DraggableCore
                  allowAnyClick={false}
                  axis="both"
                  bounds={false}
                  cancel={null}
                  defaultClassName="react-draggable"
                  defaultClassNameDragged="react-draggable-dragged"
                  defaultClassNameDragging="react-draggable-dragging"
                  defaultPosition={
                    Object {
                      "x": 0,
                      "y": 0,
                    }
                  }
                  disabled={false}
                  enableUserSelectHack={true}
                  grid={null}
                  handle={null}
                  offsetParent={null}
                  onDrag={[Function]}
                  onMouseDown={[Function]}
                  onStart={[Function]}
                  onStop={[Function]}
                  position={
                    Object {
                      "x": 286,
                      "y": 555,
                    }
                  }
                  scale={1}
                  transform={null}
                >
                  <div
                    className="tag-group animate-appear react-draggable"
                    onContextMenu={[Function]}
                    onDoubleClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    style={
                      Object {
                        "touchAction": "none",
                        "transform": "translate(286px,555px)",
                      }
                    }
                    transform={null}
                  >
                    test3
                  </div>
                </DraggableCore>
              </Draggable>
            </TagGroup>
          </div>
        </DraggableCore>
      </Draggable>
    </Canvas>
  </div>
</MappingDetail>
`;
